vault/resource_auth_backend.go

```diff
@@ -49,28 +50,35 @@ func authBackendResource() *schema.Resource {
 			},
 
 			"default_lease_ttl_seconds": {
-				Type:        schema.TypeInt,
-				Required:    false,
-				Optional:    true,
-				Computed:    true,
-				ForceNew:    true,
-				Description: "Default lease duration in seconds",
+				Type:          schema.TypeInt,
+				Required:      false,
+				Optional:      true,
+				Computed:      true,
+				ForceNew:      true,
+				ConflictsWith: []string{"tune.0.default_lease_ttl"},
```

Comment:
I'm curious why you added this as a `ConflictsWith` field. `ConflictsWith` is described as:
```
// ConflictsWith is a set of schema keys that conflict with this schema.
```
So far I've only seen it used when deprecating a different key in the schema, as described [here](https://www.terraform.io/docs/extend/best-practices/deprecations.html). So then the string used will exactly match a key in the Schema map here. 

Can you explain more about why you added this here? This question also pertains to lines 69 and 79.

Reply:
`default_lease_ttl_seconds`, `listing_visibility` and `max_lease_ttl_seconds` all overlap with fields in the ` authMountTuneSchema` that this PR is reusing from other mount's resources.

I had marked these 3 fields as deprecated (in favor of the more "standard" set of configuration for mount tuning), and utilized Conflicts with to avoid the `max_lease_ttl_seconds` being set in both this top level block and in the tune block, which would cause an issue with one value winning over the other and/or the configuration flapping between the two. 

My understanding is that this is what is recommended when safely renaming an optional field (https://www.terraform.io/docs/extend/best-practices/deprecations.html#renaming-an-optional-attribute), which is effectively what this PR is doing, I think, but if that was a misunderstanding on my part I'm happy to readjust how that is done.

Reply:
That makes sense! Thank you!